lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -1042,68 +1047,85 @@ namespace Js
         }
         uint32 growSize = newBufferLength - this->bufferLength;
 
-        bool failedReport = false;
-        const auto reportFailedFn = [&failedReport] { failedReport = true; };
+        struct AutoFailFastOnBadState
+        {
+            bool isStateValid = true;
+            ~AutoFailFastOnBadState()
```

Comment:
I'm not sure I understand your point. What is the link between QueryContinue and WebAssembly.GrowMemory ?
After we've called ReAlloc, `this->buffer` is freed. Which means, any kind of exception before we detach the buffer will leave us in a bad state and there isn't much we can do to recover at that point. 
This should be a very rare scenario, most of the time in very intense memory usage and thus it makes sense to failfast.